### PR TITLE
INT-4593 Ingest Google Secret Manager

### PR DESCRIPTION
**Pull request by i5o**
- [x] ingest google secret manager
   - [x] add fetch secrets step
   - [x] add fetch secret versions steps
   - [x] update md documentation
   - [x] add tests

### CODE REVIEW

**Comment by i5o** [comment]
Moved to review. It's still missing the test but would appreciate feedback on existing code

File: src/steps/secret-manager/index.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  await jobState.iterateEntities(
+    { _type: ENTITY_TYPE_SECRET_MANAGER_SECRET },
+    async (secretEntity) => {
+      const secret =
+        getRawData<secretmanager_v1beta1.Schema$Secret>(secretEntity);
+
+      // TO-DO: Why is this necessary?
+      if (!secret) return;
+
+      await client.iterateSecretVersions(secret, async (version) => {
```
Comment (ndowmon) [inline src/steps/secret-manager/index.ts:53]: Per your comment above, you can instead assert that `secret` is present:


```suggestion
      await client.iterateSecretVersions(secret!, async (version) => {
```

**Review by ndowmon — COMMENTED** [review]
Thanks for opening this PR, @i5o! This looks good for your first PR, nice job. I've posted a few requested changes.

File: test/recording.ts [hunk 2]
```diff
@@ -244,11 +244,17 @@ function redact(entry): void {
...
       entry.response.content.text = JSON.stringify(getRedactedOAuthResponse());
       return;
     }
 
-    responseText = gzipStringToUtf8(responseText);
+    responseText = gzipStringToUtf8(
+      responseText,
+      entry.response.content.encoding,
+    );
+    entry.response.content.encoding = 'utf-8';
```
Comment (ndowmon) [inline test/recording.ts:256]: Thanks so much for your massive effort debugging the Polly.js recordings! You really had to drink from the firehose on this one, and you came out on top. Well done! 💪

File: src/steps/secret-manager/index.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            from: secretEntity,
+            to: versionEntity,
+          }),
+        );
+      });
+    },
+  );
+}
+
+export const secretManagerSteps: IntegrationStep<IntegrationConfig>[] = [
```
Comment (ndowmon) [inline src/steps/secret-manager/index.ts:61]: These integration steps look great. Well done.

**Review by ndowmon — APPROVED** [review]
Thanks for working through issues and bringing this project into the current age. Let's discuss how to make the docs & development experience the best that they can be.

**Pull request merged**